pkg/utils/ipset.go

```diff
@@ -441,45 +472,22 @@ func (set *Set) Swap(setTo *Set) error {
 
 // Refresh a Set with new entries.
 func (set *Set) Refresh(entries []string, extraOptions ...string) error {
-	var err error
-	// The set-name must be < 32 characters!
-	tempName := set.Name + "-"
-
-	newSet := &Set{
-		Parent:  set.Parent,
-		Name:    tempName,
-		Options: set.Options,
-	}
-
-	err = set.Parent.Add(newSet)
-	if err != nil {
-		return err
-	}
-
-	for _, entry := range entries {
-		_, err = newSet.Add(entry)
-		if err != nil {
-			return err
-		}
-	}
-
-	err = set.Swap(newSet)
-	if err != nil {
-		return err
-	}
+	entriesWithOptions := make([][]string, len(entries))
 
-	err = set.Parent.Destroy(tempName)
-	if err != nil {
-		return err
+	for index, entry := range entries {
+		entriesWithOptions[index] = append([]string{entry}, extraOptions...)
```

Comment:
Note that this, technically, changes the `Refresh` behavior. The `extraOptions` param was ignored earlier, but it’s properly handled now.